### DISCUSSION TITLE
Inconsistent use of 0-indexing and 1-indexing

### DISCUSSION THREAD

**Discussion by kaibert** [FakeFishGames/Barotrauma#14993] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The circuit-box is 1-indexed (already not good), i.e. [ SIGNAL_IN_1, SIGNAL_IN_2, ... ]
but the internal components are 0-indexed, for example the input selector (mux), i.e. [SIGNAL_IN_0, SIGNAL_IN_1, ... ].

Inconsistent indexing of arrays is not good. The correct industry practice is to always use 0-indexing [ 0, 1, 2, 3, ... ].

### Reproduction steps

1) Create a circuit box
2) Add an input_selector
3) Be slightly annoyed that you have to connect signal_in_1 to signal_in_0 in the input_selector and then keep track of this mess.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.6.18.1 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
Also tracks in jukebox are 0-3, instead of track name or at least 1-4

**Reply by NoahRoopull** [reply]
I believe that's due to how the ManuallySelectedSound interface works. Sounds are 0-indexed (because, you know, C#), but with a few status effects in the IntegerInput, displaying song name in the label (e.g. "Alien Syndicate" instead of "Track") is doable with mods.

**Comment by Jasontti** [comment]
Inconsistency is never a good thing.
For wiring i would go with 1-indexing, as that's what is used in actual circuit boxes and other electrical installations.
Also it's more apt for most of the cases that count actual things.

**Comment by 3e849f2e5c** [comment]
Added 4 new pins to the circuit box, SIGNAL_IN_0, SIGNAL_IN_9, SIGNAL_OUT_0 and SIGNAL_OUT_9
